06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Inverted Index class
+ */
+class InvertedIndex {
+/**
+ * class constructor
+ */
+  constructor() {
+    this.indices = {};
+    this.documentCount = 0;
+  }
+/**
+ * normalizeText takes a string with numbers,
+ * spaces and symbols and returns a normalized string
+ * @param{String} text - The name of the string to be normalized
+ * @return{Array} Array of normalized text
+ */
+  static normalizeText(text) {
+    return (text.toLowerCase().match(/[A-Za-z]+/g).sort());
+  }
+/**
+ * @param{String} words - String to be filtered
+ * @return{Array} token - Array without duplicate words
+ */
+  static uniqueWords(words) {
+    const tokens = InvertedIndex.normalizeText(words);
+    return tokens.filter((item, index) =>
+                tokens.indexOf(item) === index);
+  }
+  /**
+   * isValidFile checks the validity of uploaded files
+   * @param {Object} file the selected file
+   * @returns {Boolean} check - returns true for valid files
+   * and false for invalid files.
+   */
+  isValidFile(file) {
+    this.file = file;
```

Comment:
Any reason why you doing this? You are already passing 'file', why reassigning to this.file. If this is to avoid "class-methods-use-this" error, you can make isValidFile static. Give it a thought.